sphinx/environment/__init__.py

```diff
@@ -647,6 +667,18 @@ def prepare_settings(self, docname):
         if 'smart_quotes' not in self.settings:
             self.settings['smart_quotes'] = True
```

Comment:
This should follow `smartquotes` confval. It's my mistake. I'll fix this later.

Reply:
yes, 1.6.6 has `self.settings['smart_quotes'] = self.config.smartquotes`, sometimes "merge" has surprises `:-)`